src/6model/serialization.c

```diff
@@ -337,21 +337,9 @@ void MVM_serialization_write_array(MVMThreadContext *tc,
     }
 }
 
-/* Writing function for null-terminated char array strings */
-void MVM_serialization_write_cstr(MVMThreadContext *tc, MVMSerializationWriter *writer, char *string) {
-    size_t len;
-    if (string)
-        len = strlen(string);
-    else
-        len = 0;
-    if (len) {
-        MVM_serialization_write_int(tc, writer, len);
-        expand_storage_if_needed(tc, writer, len);
-        memcpy(*(writer->cur_write_buffer) + *(writer->cur_write_offset), string, len);
-        *(writer->cur_write_offset) += len;
-    } else {
-        MVM_serialization_write_int(tc, writer, 0);
-    }
+/* Writing function for null-terminated char array strings. */
+void MVM_serialization_write_cstr(MVMThreadContext *tc, MVMSerializationWriter *writer, const char *string) {
+    MVM_serialization_write_array(tc, writer, string, string && strlen(string));
```

Comment:
Looks like a Perlism slipped through. I'm pretty sure this will pass 1 instead of the string's length as `size` and if it doesn't the compiler actually violates the C standard.